client/ayon_maya/api/lib.py

```diff
@@ -4544,3 +4557,83 @@ def get_scene_units_settings(project_settings=None)-> tuple[str, str]:
     linear_unit = scene_units.get("linear_units", "cm")
     angular_unit = scene_units.get("angular_units", "deg")
     return linear_unit, angular_unit
+
+
+def apply_connections(connections, target_nodes_by_id, source_nodes_by_id):
+    """Connect texture reference object nodes to the target object
+    nodes if there is one.
+
+    Args:
+        connections (list): list of texture reference
+        objects connection data
+        target_nodes_by_id (dict): The dict with target shape node ids
+        source_nodes_by_id (dict): The dict with texture reference node ids.
```

Comment:
This may identify type issues better.
```suggestion
        target_nodes_by_id (dict[str, list[str]]): The dict with target shape node ids
        source_nodes_by_id (dict[str, list[str]]): The dict with texture reference node ids.
```